src/core/infer_trace.h

```diff
@@ -69,6 +69,10 @@ class InferenceTrace {
   void Report(
       const TRITONSERVER_InferenceTraceActivity activity, uint64_t timestamp_ns)
   {
+    if (level_ < TRITONSERVER_TRACE_LEVEL_TIMESTAMPS) {
```

Comment:
Do not use '<'. level_ is a bitmask of enabled trace levels. So you need to mask TRACE_LEVEL_TIMESTAMPS and only report if that bit is set.